From foo@baz Tue Apr  9 12:12:43 2002
Date: Wed, 13 Aug 2008 20:08:35 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: USB gadget: add u_ether.c to g_util.ko

As u_ether.c is also included in multiple .ko files, move it into
g_util.ko

Cc: David Brownell <david-b@pacbell.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/Makefile  |    6 +++---
 drivers/usb/gadget/u_ether.c |    5 ++++-
 2 files changed, 7 insertions(+), 4 deletions(-)

--- a/drivers/usb/gadget/Makefile
+++ b/drivers/usb/gadget/Makefile
@@ -24,16 +24,16 @@ obj-$(CONFIG_USB_M66592)	+= m66592-udc.o
 #
 
 g_utils-objs			:= composite.o usbstring.o config.o \
-					epautoconf.o
+					epautoconf.o u_ether.o
 
 g_zero-objs			:= zero.o f_sourcesink.o f_loopback.o
-g_ether-objs			:= ether.o u_ether.o f_subset.o f_ecm.o
+g_ether-objs			:= ether.o f_subset.o f_ecm.o
 g_serial-objs			:= serial.o u_serial.o f_acm.o f_serial.o
 g_midi-objs			:= gmidi.o
 gadgetfs-objs			:= inode.o
 g_file_storage-objs		:= file_storage.o
 g_printer-objs			:= printer.o
-g_cdc-objs			:= cdc2.o u_ether.o f_ecm.o \
+g_cdc-objs			:= cdc2.o f_ecm.o \
 					u_serial.o f_acm.o
 
 ifeq ($(CONFIG_USB_ETH_RNDIS),y)
--- a/drivers/usb/gadget/u_ether.c
+++ b/drivers/usb/gadget/u_ether.c
@@ -797,6 +797,7 @@ int __init gether_setup(struct usb_gadge
 
 	return status;
 }
+EXPORT_SYMBOL_GPL(gether_setup);
 
 /**
  * gether_cleanup - remove Ethernet-over-USB device
@@ -817,7 +818,7 @@ void gether_cleanup(void)
 
 	the_dev = NULL;
 }
-
+EXPORT_SYMBOL_GPL(gether_cleanup);
 
 /**
  * gether_connect - notify network layer that USB link is active
@@ -891,6 +892,7 @@ fail0:
 		return ERR_PTR(result);
 	return dev->net;
 }
+EXPORT_SYMBOL_GPL(gether_connect);
 
 /**
  * gether_disconnect - notify network layer that USB link is inactive
@@ -962,3 +964,4 @@ void gether_disconnect(struct gether *li
 	link->ioport = NULL;
 	spin_unlock(&dev->lock);
 }
+EXPORT_SYMBOL_GPL(gether_disconnect);
